code/components/citizen-scripting-core/src/ResourceScriptingComponent.cpp

```diff
@@ -78,6 +80,15 @@ ResourceScriptingComponent::ResourceScriptingComponent(Resource* resource)
 					}
 				}
 
+				for (auto& sharedScript : sharedScripts)

```

Comment:
Code duplication is annoying. It's probably better to make an outer loop that iterates over lists, such as:

```cpp
for (auto& list : { blahScripts, sharedScripts })
{
    for (auto& script : list)
    {
        // ...
    }
}
```